go/vt/vtctl/workflow/server.go

```diff
@@ -2145,6 +2174,9 @@ func (s *Server) buildTrafficSwitcher(ctx context.Context, targetKeyspace, workf
 			}
 		}
 	}
+	if wopts.ignoreSourceKeyspace {
+		return ts, nil
+	}
```

Comment:
We don't end up setting `ts.isPartialMigration()`.  This might cause an issue for any other logic that depends on this flag.

Reply:
I would note that:
1. If the source keyspace has been deleted then there's no way for us to know if it was partial
2. We don't care at this point in cancel/complete because we're ignoring the source keyspace and, whether it was partial or not, we have our migration targets that we'll be working with

Am I missing something?

Reply:
The only potential issue I can think of is leaving orphaned keyspace routing rules in place. I'll confirm that and think about potential ways to avoid that in this scenario.

Reply:
Right, so currently the way to determine partial migrations requires the source keyspace. Now, with workflow options, we would have set that in the options map I guess. We can do that in the future if we need some major new feature for partial migrations.

Maybe for now, if it easier, we can just return a message that they need to clean up the shard routing rules manually. This feature will probably be used very rarely for shard-by-shard migrations (since they are so rare to start with) I would think!